docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -384,6 +384,60 @@ Since your `ComWrapper` subclass was designed to support `CreateObjectFlags.Uniq
 (rcw as IDisposable)?.Dispose();
 ```
 
+## COM Activation with `ComWrappers`
+
+The creation of COM objects is typically performed via COM Activation &ndash; a complex scenario outside the scope of this document. In order to provide a conceptual pattern to follow we introduce the [`CoCreateInstance()`][api_cocreateinstance] API, used for COM Activation, and illustrate how it can be used with `ComWrappers`.
+
+Assume you have the following C# code in your application. The below example uses `CoCreateInstance()` to activate a COM class and the built-in COM interop system to marshal the COM instance to the appropriate interface.
+
+```csharp
+public static I ActivateClass<C, I>() where C : class
+{
+    Guid clsid = typeof(C).GUID;
+    Guid iid = typeof(I).GUID;
+    int hr = CoCreateInstance(ref clsid, IntPtr.Zero, /*CLSCTX_INPROC_SERVER*/ 1, ref iid, out object obj);
+    if (hr < 0)
+    {
+        Marshal.ThrowExceptionForHR(hr);
+    }
+    return (I)obj;
+}
+
+[DllImport("Ole32")]
+extern static int CoCreateInstance(
+    ref Guid rclsid,
+    IntPtr pUnkOuter,
+    int dwClsContext,
+    ref Guid riid,
+    [MarshalAs(UnmanagedType.Interface)] out object ppObj);
+```
+
+Converting the above to use `ComWrappers` involves removing the `MarshalAs(UnmanagedType.Interface)` from the `CoCreateInstance()` P/Invoke and performing the marshalling manually. The [`ComWrappers.GetOrRegisterObjectForComInstance()`][api_comwrappers_getorregisterobjectforcominstance] could also be used to abstract away factory style functions and activate from within a class constructor.
+
+```csharp
+static ComWrappers s_ComWrappers = ...;
+
+public static I ActivateClass<C, I>() where C : class
+{
+    Guid clsid = typeof(C).GUID;
+    Guid iid = typeof(I).GUID;
+    int hr = CoCreateInstance(ref clsid, IntPtr.Zero, /*CLSCTX_INPROC_SERVER*/ 1, ref iid, out IntPtr obj);
+    if (hr < 0)
+    {
+        Marshal.ThrowExceptionForHR(hr);
+    }
+    return (I)s_ComWrappers.GetOrCreateObjectForComInstance(obj, CreateObjectFlags.None);
+}
+
+[DllImport("Ole32")]
+extern static int CoCreateInstance(
+    ref Guid rclsid,
+    IntPtr pUnkOuter,
+    int dwClsContext,
+    ref Guid riid,
+    out IntPtr ppObj);
+```
+
 ## Additional considerations
 
 **Native AOT** &ndash; Ahead-of-time (AOT) compilation provides improved startup cost as JIT compilation is avoided. Removing the need for JIT compilation is also often required on some platforms. Supporting AOT was a goal of the `ComWrappers` API, but any wrapper implementation must be careful not to inadvertently introduce cases where AOT breaks down, such as using reflection. The `Type.GUID` property is an example of where reflection is used, but in a non-obvious way. The `Type.GUID` property uses reflection to inspect the type's attributes and then potentially the type's name and containing assembly in order to generate its value.
```

Comment:
Maybe worth noting that reflection of this kind on `typeof` arguments should be trim-compatible?

Reply:
Good point. I updated the example.